_posts/2022-12-19-tm1-project.md

```diff
@@ -0,0 +1,222 @@
+  ],
+  "Tasks":
+  {
+    "export_before_push":
+    {
+      "Process": "Processes('application_push_wrapper')"
+    },
+    "import_after_pull":
+    {
+      "Process": "Processes('application_pull_wrapper')"
+    },
+    "save_data":
+    {
+      "Process": "Processes('application_save_data_all')"
+    },
+    "backup_instance":
+    {
+      "Process": "Processes('application_create_backup')"
+    },
+    "post_deployment_task":
+    {
+      "Process": "Processes('application_post_deployment')"
+    }
+  },
+  "PrePush":
+  [
+    "Tasks('export_before_push')"
+  ],
+  "PrePull":
+  [
+    "Tasks('save_data')",
+    "Tasks('backup_instance')"
+  ],
+  "PostPull":
+  [
+    "Tasks('import_after_pull')",
+    "Tasks('post_deployment_task')"
+  ],
+  "Deployments": 
+  {
+      "DEV":
+      {
+          "Settings":
+          {
+              "Administration":
+              {
+                  "PerformanceMonitorOn": true,
+                  "PerfMonActive": true,
+                  "TopLog":
+                  {
+                      "Enable": true
+                  }
+              }
+          }
+      },
+      "QA":
+      {
+          "Settings":
+          {
+              "Administration":
+              {
+                  "PerformanceMonitorOn": true,
+                  "PerfMonActive": true,
+                  "TopLog":
+                  {
+                      "Enable": true
+                  }
+              }
+          }
+      },
+      "PROD":
+      {
+          "Settings":
+          {
+              "Administration":
+              {
+                  "PerformanceMonitorOn": false,
+                  "PerfMonActive": false,
+                  "TopLog":
+                  {
+                      "Enable": true
+                  }
+              }
+          }
+      }
+   }
+}
+```
+
+
+
+#### What would this tm1project do?
+
+- [ ] It will push all files under the git_source_files/data **AND** git_source_files/scripts to the repo
+- [ ] It will push attributes and security objects (not their data)
+- [ ] It will push cubes, dimensions, hierarchies, views, subsets, processes and chores (unless specified in the [Ignore] section)
+- [ ] It will ignore processes that start with 'zTest' and 'WIP' (you can add more wildcards in subsequent lines)
+- [ ] It will run 'export_before_push' before pushing to the repo (thus ensuring source files are up-to-date)
+- [ ] It will run 'save_data' and 'backup_instance' processes before pulling (thus modifying) anything in the target tm1 db
+- [ ] It will run 'import_after_pull' immediately after updating the model and source files from the repo
+- [ ] It will run 'post_deployment_task' to complete the deployment with any custom processes needed (i.e.: specific to a realease)
+- [ ] It will update the tm1s.cfg file (for 'PerformanceMonitorOn', 'PerfMonActive' and 'TopLog' fields) depending on the deployment (environment) in which it is run.
+
+## Why should you consider it?
```

Comment:
I'm glad you included this section because it's not obvious IMO why TM1 devs should use TM1 git